### DISCUSSION TITLE
`winget` can uninstall applications that shouldn't be uninstallable

### DISCUSSION THREAD

**Discussion by casperiv0** [microsoft/winget-cli#1065] [Ideas]
# Brief description of your issue

Some apps that cannot be uninstalled via the Windows Settings App/Control Panel, can be uninstalled via winget (that probably also shouldn't be uninstallable.). 

# Steps to reproduce

1. Find an application that cannot be uninstalled via the Windows Settings App
2. Find the application in winget using `winget list`
3. Run `winget uninstall <application_name_id>`
4. The application will successfully uninstalled/not usable anymore
5. The application will still be listed in the settings but cannot be opened anymore because it can't find the correct files.

## To fix

1. Open settings application
2. Find the application in the Apps section
3. Click advanced options
4. Reset/repair the application
5. Application should be working again (but can be uninstalled again via winget)

## tested applications

_These application were uninstalled via winget but could not be uninstalled via the Settings app_

- Cortana
- Windows Store
- Windows Alarms and clocks
- Microsoft Edge 
- Windows Camera
- XboxGameOverlay
- Your phone

## Screenshot

Uninstalls the application

![image](https://user-images.githubusercontent.com/53900565/120218721-b28d8980-c229-11eb-8091-c455cfdb7e7b.png)

Application cannot be opened anymore

![image](https://user-images.githubusercontent.com/53900565/120218866-e072ce00-c229-11eb-8d26-f2abc97c191f.png)

# Expected behavior

Should not uninstall apps that cannot be uninstalled normally (via Control panel/Windows Settings app)

# Actual behavior

It's uninstalling the application that should not be uninstallable

# Environment

```none
[winget --info]
Windows Package Manager version: v1.0.11451
Windows: Windows.Desktop v10.0.21390.1
Package: Microsoft.DesktopAppInstaller version: v1.11.11451.0

Any other software?

- Windows insiders (dev channel)
```

**Comment by danielehrhardt** [comment]
I like the Possibility to uninstall the Windows Store.

**Comment by denelon** [comment]
I'm going to convert this to a discussion. Several others have requested the ability to remove any package including those hidden in Add / Remove Programs. There are other ways like Remove-AppxPackage in PowerShell, that should also be addressed if this is something we need to build support around.

**Reply by casperiv0** [reply]
To be honest, it would be cool to keep this feature :D. I created the issue because I wasn't sure if it was intentional that such application could be uninstallable 😅.
